src/components/Carousel.js

```diff
@@ -89,7 +89,7 @@ class Carousel extends Component {
     }
 
     componentWillReceiveProps (nextProps) {
-        if (nextProps.selectedItem !== this.state.selectedItem) {
+        if (nextProps.selectedItem !== this.props.selectedItem) {
```

Comment:
Could you elaborate on why you think this is the right solution?

Reply:
there seems to be a race condition problem when setting state `onChange`. I think it's because react batches `setState` calls to improve performance, and I see that none of the `setState` in the Carousel are happening in `setState`'s callback function. Because of `setState`'s async nature, this results in unwanted state.
I found that when comparing `nextProps` to `props`, the error that I'm facing disappears. So this might be a quick fix until the core issue of async setState is fixed?
That being said, your example code does work, because you add this check
`if (currentSlide !== index) {`
which works for me too... But that doesn't fix the core problem.
Thanks for your response.